app/models/setting.rb

```diff
@@ -45,6 +45,7 @@ def validate(record)
   validates :value, :array_hostnames_ips => true, :if => proc { |s| ARRAY_HOSTNAMES.include? s.name }
   validates :value, :email => true, :if => proc { |s| EMAIL_ATTRS.include? s.name }
   before_save :clear_value_when_default
+  before_save :encrypt_url_if_password_present, :if => proc { |s| s.value.to_s.start_with?('http') }
```

Comment:
I'd save the time and only validate strings. Possibly look at the setting's definition if it's supposed to be a string.
```suggestion
  before_save :encrypt_url_if_password_present, :if => proc { |s| s.value.is_a?(String) && s.value.start_with?('http') }
```